### PR TITLE
Czech language: initial support

### PR DESCRIPTION
**Pull request by Tony763**
Initial support. All test passed. Will need future improvments.

### CODE REVIEW

File: lingua_franca/lang/parse_cz.py [hunk 1]
```diff
@@ -0,0 +1,1647 @@
...
+                used += 1
+                start -= 1
+                # parse 15 of July, June 20th, Feb 18, 19 of February
+        elif word in months or word in monthsShort and not fromFlag:
+            try:
+                m = months.index(word)
+            except ValueError:
+                m = monthsShort.index(word)
+            used += 1
+            datestr = _MONTHS_CONVERSION.get(m)  # Convert czech months to english
```
[the remark this answers is not in the document]
Comment (Tony763) [reply lingua_franca/lang/parse_cz.py]: > What's the benefit in converting these to the English month names?
At line 1451: you originally supply `datestr ` to `datetime.strptime` and it only know months in english

File: lingua_franca/format.py [hunk 2]
```diff
@@ -504,6 +504,7 @@ def nice_duration(duration, lang=None, speech=True):
                 out += _translate_word("day", lang)
             else:
                 out += _translate_word("days", lang)
+            out += ""
```
[the remark this answers is not in the document]
Comment (Tony763) [reply lingua_franca/format.py:507]: If there is space inside you actually need a second space in test string in English duration test, without it, it pass normally with one space between words:

```----------------------------------------------------------------------
Traceback (most recent call last):
  File "w:\lingua-franca\test\test_format.py", line 584, in test_nice_duration
    self.assertEqual(nice_duration(500000),
AssertionError: 'five days  eighteen hours fifty three minutes twenty seconds' != 'five days eighteen hours fifty three minutes twenty seconds'
- five days  eighteen hours fifty three minutes twenty seconds
?          -
+ five days eighteen hours fifty three minutes twenty seconds
```

File: lingua_franca/format.py [hunk 3]
```diff
@@ -208,7 +212,7 @@ def date_format(self, dt, lang, now):
             weekday=self.lang_config[lang]['weekday'][str(dt.weekday())],
             month=self.lang_config[lang]['month'][str(dt.month)],
             day=self.lang_config[lang]['date'][str(dt.day)],
-            formatted_year=self.year_format(dt, lang, False))
+            formatted_year=date_time_format.year_format(dt, lang, False))
```
[the remark this answers is not in the document]
Comment (Tony763) [reply lingua_franca/format.py]: done

File: lingua_franca/format.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tony763) [reply lingua_franca/format.py:507]: for czech it generate the same error:

`Traceback (most recent call last):
  File "w:\lingua-franca\test\test_format_cs.py", line 609, in test_nice_duration
    self.assertEqual(nice_duration(500000,),
AssertionError: 'pět dní  osmnáct hodiny padesát tři minuty dvacet sekundy' != 'pět dní osmnáct hodiny padesát tři minuty dvacet sekundy'
- pět dní  osmnáct hodiny padesát tři minuty dvacet sekundy
?        -
+ pět dní osmnáct hodiny padesát tři minuty dvacet sekundy`

**Comment by Tony763** [comment]
yes some help would be good. I replied to all your questions (dont know if i have or not to click on resolve conversation), put all missing files back, but i cant find a way to squash commits. Do I have to close this PR and make a new one with right settings?

File: test/test_format.py [hunk 4]
```diff
@@ -582,7 +582,7 @@ def test_nice_duration(self):
                          "thirteen hours fifty three minutes twenty seconds")
         self.assertEqual(nice_duration(50000, speech=False), "13:53:20")
         self.assertEqual(nice_duration(500000),
-                         "five days  eighteen hours fifty three minutes twenty seconds")  # nopep8
+                         "five days eighteen hours fifty three minutes twenty seconds")  # nopep8
```
Comment (ChanceNCounter) [inline test/test_format.py:585]: On the one hand, `#nopep8` makes me think this test was meant to verify that `nice_duration()` works with odd spaces in the input string, so this shouldn't be fixed.

On the other hand, the fact that the tokenizers work is proof positive that whitespace doesn't matter here, so `#nopep8` could  come out.

Either way, not really relevant to this PR, but it's the kind of thing that linters often catch automatically so you merge it anyway.

**Comment by ChanceNCounter** [comment]
> yes some help would be good. I replied to all your questions (dont know if i have or not to click on resolve conversation), put all missing files back, but i cant find a way to squash commits. Do I have to close this PR and make a new one with right settings?

Whoever merges the PR can squash it at that time. If you'd like to learn about squashing it yourself, it's `git rebase` that you want to investigate. I'd make a copy of your branch and use it to experiment, as most people screw up their first few rebases. However, it's a useful skill, so I'd encourage you to check it out.

Regardless, somebody else can take care of it in the GitHub interface.

**Comment by Tony763** [comment]
@ChanceNCounter, @krisgesling   -  rebased

**Comment by Tony763** [comment]
Do I have to do something more?

**Comment by ChanceNCounter** [comment]
Nope, sorry. I was trying to suss out whether that refactor is about to be merged, because it'd make about half this stuff unnecessary, and this would be the easier PR to alter.

But it doesn't look like that PR is gonna be merged today or tomorrow, so if nobody else gets to this by the time I'm at a computer (I'm on my phone) I'll re-review and potentially merge.

File: lingua_franca/format.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChanceNCounter) [reply lingua_franca/format.py:507]: Confirmed: the error in `nice_duration()` was masked by the same error in the English-language test. At first glance, I thought that test was verifying the normalizer, but, nope. Good catch.

**Comment by ChanceNCounter** [comment]
Well, I've looked it over, and everything seems to be in the right place. You should delete the `.egg`, don't wanna pull binaries, but I don't see anything else right away. I usually wait a couple hours and look again to be sure, though. Good devs don't trust themselves 😛 

It also occurs to me that, not speaking Czech, I would have no way of knowing if there's an egregious spelling or grammatical error in there, or even if it's just a bunch of profanity. While you're deleting that .eggfile, we should ask @krisgesling whether Mycroft's had a solution to that in the past.

**Comment by Tony763** [comment]
> You should delete the `.egg`, don't wanna pull binaries

Removed, rebased. Shouldn't we add dist and egg folder to .gitignore?

File: test/test_format.py (hunk 4 above)
Comment (krisgesling) [reply test/test_format.py:585]: I'd agree on this one. As it's the only test in the file with a `#nopep8` flag I think it's intentionally there to test odd spacing.
@Tony763 can you switch this back please?

**Comment by krisgesling** [comment]
Yeah my Czech isn't great either. 

I kind of expect that something isn't perfect but as you said elsewhere one of the easiest ways to find out is to get it out in peoples hands. Once we have a few Czech speakers using it, any inconsistencies should get flagged pretty quickly. 

The biggest concern I guess would be someone intentionally adding inappropriate content in another language however they'd be going to a lot of effort for a very small joke, it would also likely get picked up fairly quickly by another speaker of that language, and we don't officially support other languages yet so we're not making any guarantees as to the accuracy of things.

I think this is ready to merge as soon as we revert the tiny change on the `#nopep8` test :tada:

**Comment by Tony763** [comment]
Space reverted back. As You say, I know this is imperfect and that is why I named it initial. It will by easier to lure some czech devs or czech/english speaking people that can help with translation of skills.  Really only few people want to translate something when they can't use it.

And You will be first with working (beta) voice assistant, because none of big players have it (Google, Amazon, Apple)

**Comment by krisgesling** [comment]
Wow that is very cool, I didn't realise just how momentous this was! 

:rocket: :rocket: :rocket: :rocket: Merging :rocket: :rocket: :rocket: :rocket:

**Comment by krisgesling** [comment]
Also if you'd like a dedicated Czech language channel on Mycroft Chat, please [let me know](https://chat.mycroft.ai/community/messages/@gez-mycroft).

**Comment by Tony763** [comment]
Firstly I will send more PRs with finished translations I have made with @manasovo

**Pull request merged**